front-end/src/util/api.js

```diff
@@ -10,6 +10,13 @@ export function api ({
         payload.body = JSON.stringify(body)
     }
     return fetch(endpoint, payload)
-        .then(response => response.json())
-        .catch(error => toast.error(error.message));
+        .then(response => {
+            if (response.status === 409){
+                toast.error("Already exists");
+            }
+            else if (!response.ok) {
+                toast.error("Unexpected Error");
+            }
+            return response.json();
```

Comment:
We should probably handle these case outside the function. Otherwise, we won't be able to customize the error message.